src/main/scala/com/snowplowanalytics/indicative/FieldsExtraction.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2018-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.indicative
+
+// Java
+import java.time.ZonedDateTime
+
+// Scala
+import scala.annotation.tailrec
+
+// cats
+import cats.syntax.either._
+
+// circe
+import io.circe.Json
+
+// Analytics SDK
+import com.snowplowanalytics.snowplow.analytics.scalasdk.json.Data
+import com.snowplowanalytics.snowplow.analytics.scalasdk.json.JsonShredder
+
+// This library
+import com.snowplowanalytics.indicative.Transformer.TransformationError
+
+object FieldsExtraction {
+
+  /**
+   * Gets a json value out of the map and changes it to string, which is what the Indicative API expects
+   */
+  def extractField(properties: Map[String, Json], key: String): Either[TransformationError, String] =
+    properties.get(key).flatMap(_.asString).toRight(TransformationError(s"Could not find the field with key $key"))
+
+  /**
+   * Converts a timestamp string with timezone into milliseconds
+   */
+  def timestampToMillis(timestampString: String): Either[TransformationError, Long] =
+    Either
+      .catchNonFatal(ZonedDateTime.parse(timestampString))
+      .leftMap(error => TransformationError(s"Failed to parse the timestamp, message: ${error.getMessage}"))
```

Comment:
I think `s"Failed to parse timestamp $timestampString: ${error.getMessage}"` should be the way to go here